common/credentials/src/coconut/bandwidth/issued.rs

```diff
@@ -166,22 +166,36 @@ impl IssuedBandwidthCredential {
 
     pub fn prepare_for_spending(
         &self,
-        verification_key: &VerificationKey,
+        verification_key: &VerificationKeyAuth,
+        pay_info: PayInfo,
+        coin_indices_signatures: Vec<CoinIndexSignature>,
     ) -> Result<CredentialSpendingData, Error> {
         let params = bandwidth_credential_params();
-
-        let verify_credential_request = prove_bandwidth_credential(
+        let spend_date = today_timestamp();
+        let (payment, _) = self.wallet.spend(
             params,
             verification_key,
-            &self.signature,
-            &self.serial_number,
-            &self.binding_number,
+            &self.ecash_secret_key,
+            &pay_info,
+            false,
+            constants::SPEND_TICKETS,
```

Comment:
since we're calling this on `self` and it's a hardcoded value, can't we not just remove that argument?

Reply:
The `spend` function allows a general case of spending more than one ticket. If we remove that argument, there's a whole bunch of changes we can make in the ecash lib, and it gets really purpose built